### DISCUSSION TITLE
Cannot delete Directus users - foreign keys violation

### DISCUSSION THREAD

**Discussion by gkielwasser** [directus/directus#8644] [Archive]
Hi,

I'm using RC.96, I cannot delete a user, I got this error
```
delete from "directus_users" where "id" in ($1) - update or delete on table "directus_users" violates fore
ign key constraint "directus_files_uploaded_by_foreign" on table "directus_files"

```
Does anyone has the same behavior?

Thanks

**Comment by rijkvanzanten** [comment]
@gkielwasser Looks like you have `directus_files` that are associated to the user in the uploaded_by field. Removing/nullifying those files should allow the user to be deleted. I agree that this is a little user hostile. I would expect the "uploaded by" field to be nullified on delete, instead of throw the default foreign key constraint error 👍🏻 

I'll add a new issue to get that resolved

**Reply by rijkvanzanten** [reply]
Ref https://github.com/directus/directus/issues/8650

**Reply by gkielwasser** [reply]
Ok thanks, I will follow that ticket. In the meantime I have tried to delete all files but I I have another foreign key violation on a custom collection. This collection is using "user_created" fields.
``delete from "directus_users" where "id" in ($1) - update or delete on table "directus_users" violates foreign key
 constraint "warehouses_user_created_foreign" on table "warehouses"``

**Reply by rijkvanzanten** [reply]
Same thing! Either delete those rows, or set the user_created field to null 👍🏻

**Reply by gkielwasser** [reply]
I have many collections where I have enabled the created_by/created_on fields, I will delete the linked entites manually.

But does that means in a normal process if I want to delete the linked user, I should first delete manually all those entities ?

Or the ticket will also takes care of that use case ?

**Reply by rijkvanzanten** [reply]
That's what that ticket is about 👍🏻  It will make sure that all those fields will be set to "null" when the user is deleted

**Reply by gkielwasser** [reply]
Ok perfects thanks !

**Comment by antontranelis** [comment]
it looks like this issue still exist

**Comment by mudassirwork** [comment]
any workaround for this error ?

**Comment by IanvanderGeest** [comment]
Hi @rijkvanzanten,

how does one set this constraint to `nullify`? Directus doesn't allow me to change this within the data studio because the fields of `directus_users` and `directus_files` are system fields. I also cannot find a record for this in the `directus_relations` table.

![Screenshot 2025-02-26 at 15 54 43](https://github.com/user-attachments/assets/0dc6b627-0449-4778-a2c2-5ddc35e65aee)

Reviewing the table structure, the field should be nullable
![Screenshot 2025-02-26 at 17 17 42](https://github.com/user-attachments/assets/dc1da247-8b98-4259-a87e-d3b1eaf5988e)
